File: sdk/src/main/java/io/antelli/sdk/model/AnswerItemList.java
package io.antelli.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * Handcrafted by Štěpán Šonský on 12.11.2017.
 */

class AnswerItemList implements Parcelable {

    private List<AnswerItem> items;

    public AnswerItemList(List<AnswerItem> items) {
        this.items = items;
    }

    public AnswerItemList add(AnswerItem item) {
        this.items.add(item);
        return this;
    }

    public AnswerItemList addAll(List<AnswerItem> item) {
        this.items.addAll(item);
        return this;
    }

    public List<AnswerItem> getItems() {
        return items;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.items);
    }

    protected AnswerItemList(Parcel in) {
        this.items = in.createTypedArrayList(AnswerItem.CREATOR);
    }

    public static final Creator<AnswerItemList> CREATOR = new Creator<AnswerItemList>() {
        @Override
        public AnswerItemList createFromParcel(Parcel source) {
            return new AnswerItemList(source);
        }

        @Override
        public AnswerItemList[] newArray(int size) {
            return new AnswerItemList[size];
        }
    };
}

File: sdk/src/main/java/io/antelli/sdk/model/AnswerItemList.kt
package io.antelli.sdk.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Handcrafted by Štěpán Šonský on 12.11.2017.
 */
internal class AnswerItemList : Parcelable {
    var items: MutableList<AnswerItem>
    private set

    constructor(items: MutableList<AnswerItem>) {
        this.items = items
    }

    fun add(item: AnswerItem): AnswerItemList {
        items.add(item)
        return this
    }

    fun addAll(item: List<AnswerItem>): AnswerItemList {
        items.addAll(item)
        return this
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList(items)
    }

    private constructor(parcel: Parcel) {
        items = parcel.createTypedArrayList(AnswerItem.CREATOR) ?: ArrayList()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<AnswerItemList> = object : Parcelable.Creator<AnswerItemList> {
            override fun createFromParcel(source: Parcel): AnswerItemList? {
                return AnswerItemList(source)
            }

            override fun newArray(size: Int): Array<AnswerItemList?> {
                return arrayOfNulls(size)
            }
        }
    }
}